source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -7103,6 +7108,13 @@ export class AdaptiveCard extends ContainerWithActions {
             renderedElement.style.removeProperty("minHeight");
         }
 
+        // if rtl is set to dir at the parent view of carousel, carousel's view will get disappeared
+        if (this.hasCarousel()){
+            if (this.rtl && renderedElement) {
+                renderedElement.dir = "ltl";
```

Comment:
shouldn't this be:

```suggestion
                renderedElement.dir = "rtl";
```

Reply:
I understand that somehow having `rtl` set to "ltr" on the Swiper's parent element causes Swiper to not work correctly, and so that's why you're overriding here? Seems quite weird.

If you do need to override, then a better way to do it in a generic way is to introduce a new `forceRtlOnParentContainer(): boolean` method on `CardElement` that returns `false` by default, and override it in `Carousel` to return true.

Reply:
my comments weren't published, so the comments are out of order. @dclaux
AdaptiveCards 
           |
     Carousel
           |
  CarouselPage
if we set rtl at AdaptiveCard level, the carousel will disappear, so I'm trying to override the rtl of AdaptiveCards, not the rtl of Carousel.

Reply:
@jwoo-msft  that's precisely what I'm saying.

Your code in AdaptiveCard.internalRender should read

```typescript
protected internalRender(): HTMLElement | undefined {
    ...
    for (let i = 0; i < this.getItemCount(); i++) {
        if (this.getItemAt(i).forceRtlOnParentContainer() && this.rtl && renderedElement) {
            renderedElement.dir = "rtl";
            break;
        }
    }
    ...
}
```

With `forceRtlOnParentContainer` defined in CardElement:
```typescript
export class CardElement extends CardObject {
    ...
    forceRtlOnParentContainer(): boolean {
        return false;
    }
    ...
}
```

And on Carousel:
```typescript
export class Carousel extends Container {
    ...
    forceRtlOnParentContainer(): boolean {
        return true;
    }
    ...
}
```

Reply:
I appreciate the detailed explanation, @dclaux!
i found the reason that the swiper was misbehaving with the root level, rtl. https://github.com/microsoft/AdaptiveCards/pull/6723